### PR TITLE
Delete the BlockStore trait, move BlockStoreExt, and delete forest_ipld_blockstore 

### LINKED ISSUE #2085: Delete the `BlockStore` trait, move `BlockStoreExt`, and delete `forest_ipld_blockstore`
**Issue summary**

The `BlockStore` trait is just `FVM::Blockstore + Clone` and can be removed. The `BlockStoreExt` should be renamed to `BlockstoreExt` and moved to `forest_utils::db`. Then the `forest_ipld_blockstore` crate will be empty and can be removed.

**Other information and links**

https://github.com/ChainSafe/forest/blob/d716610c32e8a7774e8b9feb8ae0f2460fe51174/ipld/blockstore/src/lib.rs#L13

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #2085

**Other information and links**

### CODE REVIEW

File: blockchain/chain_sync/src/tipset_syncer.rs [hunk 1]
```diff
@@ -1048,10 +1052,13 @@ async fn fetch_batch<DB: BlockStore + Send + Sync + 'static, C: Consensus>(
     Ok(())
 }
 
-/// Going forward along the tipsets, try to load the messages in them from the `BlockStore`,
+/// Going forward along the tipsets, try to load the messages in them from the `Blockstore + Store + Clone`,
```
Comment (lemmih) [inline blockchain/chain_sync/src/tipset_syncer.rs:1055]: This is a search&replace mistake.

File: blockchain/state_manager/src/lib.rs [hunk 2]
```diff
@@ -168,12 +170,12 @@ where
         Ok(state.get_actor(addr)?)
     }
 
-    /// Returns the cloned [`Arc`] of the state manager's [`BlockStore`].
+    /// Returns the cloned [`Arc`] of the state manager's [`Blockstore + Store + Clone`].
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:173]: More search&replace mistakes.

**Review by lemmih — CHANGES_REQUESTED** [review]
I don't think the `Store` trait is required in any of the places where you use it and the `Clone` trait can also be removed most of the time.

There are also a bunch of search&replace mistakes to be fixed.

**Review by lemmih — APPROVED** [review]
Good job. We'll get rid of those lingering `Store` trait constrains in a separate PR.

**Review by elmattic — APPROVED** [review]
Nice PR! I can still see some references to `BlockStore` in some doc comments. Maybe we could update those as well.

**Pull request merged**